File: backoffice/src/shared/infrastructure/events/DomainEventInformation.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain import DomainEvent

"""
 *
 * Class
 *
"""

class DomainEventInformation:

    """
     *
     * Attributes 
     *
    """

    __eventClass        : type
    __subscriberClasses : list[type]

    """
     *
     * Methods 
     *
    """

    def __init__( self, eventClass : type, subscriberClasses : list[type] ) -> object:
        self.__eventClass        = eventClass
        self.__subscriberClasses = subscriberClasses
    
    def getSubscriberClasses( self ) -> list[type]:
        return self.__subscriberClasses

    def getEventClass( self ) -> type:
        return self.__eventClass;

    def getEventName( self ) -> str:
        # Variables
        domainEvent : DomainEvent
        # Code
        domainEvent = self.__eventClass()
        return domainEvent.getEventName()

    def formatRabbitmqQueueName( self ) -> str:
        return 'backoffice.{}'.format( self.getEventName() );
    
    def isConsumedEvent( self ) -> bool:
        return len( self.__subscriberClasses ) > 0
    

File: preparation/src/shared/infrastructure/events/DomainEventInformation.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.shared.infrastructure {

    public class DomainEventInformation {

        /* 
         * 
         * Attributes
         *
        */

        private List<Type> subscriberClasses;
        private Type       eventClass;

        /* 
         * 
         * Methods
         *
        */

        public DomainEventInformation( 
            Type       eventClass, 
            List<Type> subscriberClasses
        ) {
            this.subscriberClasses = subscriberClasses;
            this.eventClass        = eventClass;
        }

        public List<Type> getSubscriberClasses() {
            return subscriberClasses;
        }

        public Type getEventClass() {
            return eventClass;
        }

        public string getEventName() {
            // Variables
            DomainEvent domainEvent;
            string      eventName;
            // Code
            domainEvent = ( DomainEvent )Activator.CreateInstance( eventClass );
            eventName   = domainEvent.getEventName();
            return eventName;
        }

        public string formatRabbitmqQueueName() {
            return string.Format( "preparation.{0}", getEventName() );
        }

        public bool isConsumedEvent() {
            return subscriberClasses.Count > 0;
        }

    }
}
